addon/compression_worker.h

```diff
@@ -0,0 +1,73 @@
+#ifndef COMPRESSION_WORKER_H
+#define COMPRESSION_WORKER_H
+#include <napi.h>
+
+#include <optional>
+#include <variant>
+
+using namespace Napi;
```

Comment:
It's kind of optional in projects that don't end up being included in other code, but generally using namespaces for all-but-toplevel code is a best practice (again, this is all one global namespace otherwise, if somebody happens to define a `CompressionWorker` class in another part of the application those could at least theoretically conflict)

Reply:
More like `namespace mongodb_zstd { ... }`, but yeah, that's the idea

Reply:
Done!